### ISSUE TITLE
Http2 makes downloading speed slow 

### ISSUE THREAD

**Issue by polluxau** [ValveSoftware/steam-for-linux#10248]
EDIT: This issue has been solved, if your having this issue still it may be recommended to use a newer distro for newer packages 

I had been having slow download speeds under steam on linux and was looking for a fix when i came across this reddit forum and my speeds got instantly improved 

https://www.reddit.com/r/linux_gaming/s/4NCYHM9WvP

But what could be causing this? 

Iv tried several linux distros this past week from nobara, fedora, debian, and now opensuse and they all have the same issue and disabling http2 using a .cfg in .steam/steam/steam_dev.cfg fixes the issue on all of them 

@nClientDownloadEnableHTTP2PlatformLinux 0
@fDownloadRateImprovementToAddAnotherConnection 1
@cMaxInitialDownloadSources 15

Here is a video of someone explaining it aswell and showing the improvements it made 

https://youtu.be/A_kRdad3eb4?si=5EgmSu4ZpW26aE2K

**Comment by ProjectSynchro** [comment]
Can confirm this improves speeds, although in an apples to apples test with Windows; download speeds are half of what is possible with the same hardware.

**Comment by shadywack** [comment]
Weight prioritization appears to be incorrectly set by the Linux client. Normally this is a browser request, but without knowing how the client is sending the request we're in the dark. Hopefully a Valve dev sees this and can chime in or fix it.

**Comment by Bobbydigital420** [comment]
Also experiencing this issue and have tried multiple distros also because I thought it might be an Arch linux specific problem. Lucky if I can get 3MB/s from my gigabit connection. By adding those options to the file I top out at around 40MB/s. My Windows 11 install pulls over 100MB/s. Hope this gets looked at soon, honestly surprised there's not more people talking about this.

**Comment by breadland** [comment]
The fix helps me get better speeds, although Windows 10/11 is much faster still. 650mbps minimum with my local server.

Meanwhile I gotta switch to US Boston on any Linux distro and even then it goes between 150-500mbps after adding the steam_dev.cfg fix.

**Comment by xadips** [comment]
Yup same fix helps, but I'm still missing about ~30MB/s even with the fix (110MB/s before or on windows down to ~80MB/s[measly 35MB/s without the fix])

**Comment by laurirasanen** [comment]
On my Arch setup the second line actually slows download speeds. Only the HTTP2 one is needed.

- Without cfg: 30-40mbps
- With both lines: 100mbps
- With just the first: 600mbps

This is on a 1Gbit connection in northern Europe.

**Comment by iamshadow2008** [comment]
Thank you so much. With just "@nClientDownloadEnableHTTP2PlatformLinux 0" in the .cfg my speeds went from 10MB/s to 48MB/s, which is still far off from the 74MB/s that I used to get way back when I used Windows, but it's a MASSIVE improvement. The second line: "@fDownloadRatelmprovement ToAddAnother Connection 1.0" trashed the speed again so I left it out.

**Comment by Skote2** [comment]
I'm experiencing the exact same results as @laurirasanen
1.5gbps connection in Ontario, Canada, Arch.

My steam deck is also getting the same ~30mbps download speeds as my Arch desktop. I have not tested the fix on it.

@iamshadow2008
Your issues may also be DNS related. I was observing ~10mbps before I set up a caching server.
Check the arch wiki's steam troubleshooting guide for slow download speeds [here](https://wiki.archlinux.org/title/Steam/Troubleshooting#Unusually_slow_download_speed) (if you're not on arch these instructions won't really help but checking the DNS resolution time of repetitive requests may help)

**Comment by Krutonium** [comment]
I can confirm that disabling HTTP2 fixed it on my NixOS desktop as well as my Steamdeck.

**Comment by Gerporgl** [comment]
I was also having slow download speeds, I'm only on linux (pop-os), can't tell about the Windows world if its faster or not.
Adding these 2 settings fixed completely my issue. Went from 15-20 MB/sec to 46MB/sec, which is my max speed (360Mbps). I haven't tried if the second setting is needed,probably not but in my case I'm still maxing out.

It seems something recent, because I have been with the same ISP and internet package for the past 4-5 years and always did my max speed when downloading with steam.

**Comment by cchulo** [comment]
is it possible for ISPs to have an effect on this issue? I had no problems on EndeavourOS for a couple of years until i had to move to a new neighborhood. With a new neighborhood usually comes a new ISP, which I started seeing this issue. The recommended fix in the OP remedied the issue. Sadly, I don't know anyone in my old neighborhood, i'd be interested to know if anyone had a similar experience to mine.

**Comment by merlinblack** [comment]
Before I did anything I was seeing exactly half of the speed I see in Windows 10. With adding dnsmasq I saw a speed increase for a time, but it dropped to half again. Turning off HTTP2, I've got around 98% of my windows speed. Happy with that - lets hope it sticks :-)

**Comment by YoinkerBoinker** [comment]
This only "improves" download speed on my end. but it still isn't able to reach the full potential i get when using Windows. There is something wrong. And all the fixes out there only seem to "improve". Though if you have a somewhat fast connection you still might notice slower speeds than possible. For example i never get my 1Gbps i only get around 700Mbps.. if i disable Https2.. if i don't disable it i will be at around 300Mbps.. while in Windows i'm at pretty much 900+Mbps. I assume lots of people who claim this fixes the issue don't have such a fast connection therwfore it already maxes speed after that fixes. 

All the workarounds that are around do not really fix the issue.
Just in case someone mentions them again:

- Disabling ipv6 - doesn't work  it even makes using Steam impossible on my End - it wont connect at all after that.
- Using dnsmasq - has no effect i litterally tried all kind of combinations - didn't get rid of the issue
- Some MTU fix that is around: No effect, don'T waste your time on this
- using the http2 fix  is the only fix that has an impact so far though it still doesn't give me fullspeed downloads

**Comment by DomiStyle** [comment]
@YoinkerBoinker I also have a Gigabit connection and with HTTP2 disabled I max out at around 900 Mbit/s, which is approximately 930 Mbit/s on the wire. Can't compare to Windows but that is the maximum my connection allows.

With HTTP2 enabled I also end up somewhere at around 300 Mbit/s.

**Comment by ArmoredCoreEnjoyer** [comment]
Hello! Sorry if this isn't helpful, but I had the same issue across every distribution I tried, which was:
Ubuntu, Arch, EndeaverOS, and Mint.
Tonight I tried Bazzite which is based off Fedora 39. It looks like the issue is fixed specifically with this distribution but I'm not sure if it's something they've done, or if it's part of something in Fedora 39. If any of you have some time, could you try Bazzite or Fedora 39 and report back if this issue is fixed for you?

It might be helpful to check and see if it's fixed this for anyone else and then maybe work from there.

**Comment by pollux56** [comment]
> Hello! Sorry if this isn't helpful, but I had the same issue across every distribution I tried, which was:
> Ubuntu, Arch, EndeaverOS, and Mint.
> Tonight I tried Bazzite which is based off Fedora 39. It looks like the issue is fixed specifically with this distribution but I'm not sure if it's something they've done, or if it's part of something in Fedora 39. If any of you have some time, could you try Bazzite or Fedora 39 and report back if this issue is fixed for you?
> 
> It might be helpful to check and see if it's fixed this for anyone else and then maybe work from there.

Bazzite has in there github page saying they applied that fix and the gui helper they have also has this fix that can be applied

**Comment by YoinkerBoinker** [comment]
> > Hello! Sorry if this isn't helpful, but I had the same issue across every distribution I tried, which was:
> > Ubuntu, Arch, EndeaverOS, and Mint.
> > Tonight I tried Bazzite which is based off Fedora 39. It looks like the issue is fixed specifically with this distribution but I'm not sure if it's something they've done, or if it's part of something in Fedora 39. If any of you have some time, could you try Bazzite or Fedora 39 and report back if this issue is fixed for you?
> > It might be helpful to check and see if it's fixed this for anyone else and then maybe work from there.
> 
> Bazzite has in their github page saying they applied that fix and the gui helper they have also has this fix that can be applied

Where is that? I checked their github page but can't seem to find it getting mentioned (or i'm just too stupid).

**Comment by tGecko** [comment]
On Windows I consistently get 90-95 Mbit/s, on linux, before the fix 20Mbit/s and now after the fix 80 Mbit/s
There's still room for improvement ;)

The fix:
`@nClientDownloadEnableHTTP2PlatformLinux 0`

**Comment by cchulo** [comment]
does using ipv4 instead of ipv6 have any impact for anyone else? I'm using OPNSense router, and had issues with ipv6 that resulted in a strange bug where some elements of steam were not loading properly. I resolved them by configuring ipv6 DNS to cloudflare's. I just recently found a setting to prefer ipv4 over ipv6 and the result is i get a small bump in even more performance (in addition to disabling HTTP/2):

![Screenshot_20240303_034053](https://github.com/ValveSoftware/steam-for-linux/assets/9343504/0b79ce2e-3edb-419b-8e71-15e1985f4373)

**Comment by mcheman** [comment]
I found another setting that has made a difference to my download speeds. Steam still isn't using the full 2.5Gbps of my connection, but now it's much closer.
|Config|Speed|
|---|---|
|Stock|**20MB/s**|
|Http2 Disable, Rate Improvement|**65MB/s**|
|Http2 Disable, Rate Improvement, Initial Sources|**170MB/s**|

My steam_dev.cfg file:
```
@nClientDownloadEnableHTTP2PlatformLinux 0
@fDownloadRateImprovementToAddAnotherConnection 1
@cMaxInitialDownloadSources 15
```

I noticed that `download_sources` in the steam console only ever showed me connected to 4 servers. 

I found that setting `@cMaxInitialDownloadSources 15` made it connect to 15 servers immediately. My download speed jumped to a max of **170MB/s** after only a few seconds, rather than the previous max of **65MB/s** it reached after 30 seconds.

**Comment by pollux56** [comment]
its weird that these things havent been applied to the steam client with an update, its another thing that some windows user might notice and go "oh linux is worse at downloading my games on steam, im going back to windows" when that isnt the case, its just certain things need to be enabled to work better. hopefully valve enables these or figures out a better solution to solve this.

**Comment by Rosentti** [comment]
Hi, author of the original Reddit post chiming in. 

If you decided to change the `@fDownloadRateImprovementToAddAnotherConnection`, you should probably set it to `1.1` or just remove it altogether, since setting it to `1` just tells it to connect to any servers, without verifying that it actually downloads faster. 
You could also try more than `1.2`, however it might connect to less servers with that value, effectively lowering download speed. 

Also of note, for some setups streaming off a single server might be faster than connecting to multiple servers, but take that advice with a grain of salt.

**Comment by masroof-maindak** [comment]
Hello, I've tried
1. Adding the two lines to my steam_dev.cfg
2. Changing download servers to the next closest one
3. dnsmasq

\- and this combination's helped me go from sub100 KBPs to sub1MBPs but it's still painfully slow.

Interestingly, when I tried downloading something on Steam a couple months back, I don't recall any discrepancy in download speed of any sort. I'm not sure what I've changed in my setup since then - if anything, but none of the three fixes I tried above have helped in any significant way.

Usually, I get ~4 MB/s.

![image](https://github.com/ValveSoftware/steam-for-linux/assets/62666332/aeec88cb-7917-402b-b091-b51f4f194449)

**Comment by YoinkerBoinker** [comment]
so i hade pretty much max for a while using the "fix". Now i'm at 500Mbps again on a connection that should be 1000Mbps. Frustrating.

**Comment by AnameVi** [comment]
Same problem and same fix on manjaro and ALT Regular Kde linux
create ~/.steam/steam/steam_dev.cfg and added lines:

@nClientDownloadEnableHTTP2PlatformLinux 0

@fDownloadRatelmprovement ToAddAnother Connection 1.0

**Comment by Milton-Ventura** [comment]
For me the fix didn't work... I can get around 90MB/s - 100MB/s on windows right at the start of the download process... With Linux i'm getting a max of 11MB/s, (ONLY 10%). 
I tried the steam_dev.conf file with all tree suggested commands, also tried them individually, sadly no success. The only improvement it did to me is that the download gets to its top speed (11MB/s) much faster now...

Downloads are slower, CS2 almost unplayable when comparing to Windows... Idk, i'm almost quiting...

**Comment by polluxau** [comment]
> For me the fix didn't work... I can get around 90MB/s - 100MB/s on windows right at the start of the download process... With Linux i'm getting a max of 11MB/s, (ONLY 10%). 
> I tried the steam_dev.conf file with all tree suggested commands, also tried them individually, sadly no success. The only improvement it did to me is that the download gets to its top speed (11MB/s) much faster now...
> 
> Downloads are slower, CS2 almost unplayable when comparing to Windows... Idk, i'm almost quiting...

The file is supposed to be a .cfg not conf 

Make sure its in 

.steam/steam/steam_dev.cfg

**Comment by Milton-Ventura** [comment]
Sorry, I misstyped, the file is a .cfg.

**Comment by marbaa** [comment]
On Windows I get constant speed 930-970Mbps. On Linux maximum 93Mbps. Neither one suggested command helped here. Difference using mentioned commands were about 2-3Mbps. Doesn't matter which location I choose.

Kubuntu 23.10

**Comment by nathub** [comment]
The second and third line don't seem to help in my small test case.
But the first line helped me the first time and then didn't help me today.  I input it via the steam console by going to steam://open/console.

I'm stuck at like 60 Mbps.
Using Fedora 40 Workstation (Intel 12400  CPU, if that helps narrow it down I dunno). Steam flatpak from Flathub. The native RPM had other issues if I remember correctly.

I'm on ethernet
Linux Kernel Version: " Linux 6.8.7-300.fc40.x86_64 "
uuhh ... MSI motherboard: MSI PRO B660M-G DDR4 LGA 1700 Intel B660
Ethernet is: Realtek® RTL8125BG 2.5Gbps LAN controller

And similarly, speeds are faster when I boot into Windows and download Steam games.
This is really frustrating, I hope Valve knows about this.

**Comment by roope242** [comment]
I'm also getting very slow download speeds on Linux.
I'm running Nobara which is based on Fedora and getting around 5MBps, download speed is very stable when measure with Speedtest server based in Helsinki so same city where Steam's download server is.
Another computer on my network running Windows gets a stable 110MBps download.

![image](https://github.com/ValveSoftware/steam-for-linux/assets/22483847/1f739ef3-e2a6-457c-b19a-4f3a7ef1d395)

Both these fixes are applied, but it looks like removing them makes no difference either.
```
roope@nobara-pc:~$ cat .local/share/Steam/steam_dev.cfg
@nClientDownloadEnableHTTP2PlatformLinux 0
@fDownloadRateImprovementToAddAnotherConnection 1.0
```
Disabling IPv6 also seems to not do anything. I'm also running dnsmasq on my local computer.

I can provide any logs if such are useful for debugging, just let me know which.

**Comment by roope242** [comment]
Replying to https://github.com/ValveSoftware/steam-for-linux/issues/10248#issuecomment-2088784167
I'm now without any apparent changes getting 110+MBps speeds again, weird.

**Comment by roope242** [comment]
Looks like my issue is only partly this one but mostly a routing issue. I'll share my findings anyway.

I need to:
Disable IPv6 globally by adding `ipv6.disable=1` to my kernel boot parameters.
Disable HTTP/2 from steam by adding `@nClientDownloadEnableHTTP2PlatformLinux 0` to my `steam_dev.cfg`.
Block `cache1-rix-retn.steamcontent.com` and `cache2-rix-retn.steamcontent.com` using my `/etc/hosts` with these lines:
```
$ tail -2 /etc/hosts
127.0.0.1 cache1-rix-retn.steamcontent.com
127.0.0.1 cache2-rix-retn.steamcontent.com
```
And only now I am getting around the same speeds as the Windows machine is.

As this looks to be kind of a mess, could you @laurirasanen try if adding the lines to your `/etc/hosts` help? If not, what ISP do you have? Mine is Lounea.

**Comment by HurricanePootis** [comment]
I too can confirm that I experience this problem. At both my house network and my university network, I never experience full speeds unless I set the following:
```
@nClientDownloadEnableHTTP2PlatformLinux 0
@cMaxInitialDownloadSources 15
```

I don't know what the issue can be, but what ever it is, it is happening on two vastly different networks. Also, as a side note, my university network did not work with ipv6 at all, and it doesn't make a difference to any of the settings I've changed; I always keep ipv6 enabled.

**Comment by TotallyNotChloe** [comment]
Having the same issue on endeavourOS, i was getting 500mbps on windows 10 but only 60mpbs on linux. 

By just cleaning the download cache i started getting 130mbps, but still far from the optimal download speeds, so i started messing around with the steam_dev.cfg.

The TL;DR of what i found for me is that `@nClientDownloadEnableHTTP2PlatformLinux 0` give double the speed but makes it unstable, adding `@fDownloadRateImprovementToAddAnotherConnection 1` make it more stable but reduces the overall speed. 


I tested the configs using a 20gb game, here the time it took:

After cleaning the cache: 20 minutes.

With only `@nClientDownloadEnableHTTP2PlatformLinux 0` =  7 Minutes (300mbps-473mbps), but it was really unstable, the download would stop for 1-3 seconds some times.

`@nClientDownloadEnableHTTP2PlatformLinux 0 + @fDownloadRateImprovementToAddAnotherConnection 1` = 17 minutes(100mbps-473mbps), it was stable but overall way slower, so i prefer keeping only the HTTP line.

**Comment by YoinkerBoinker** [comment]
I consider the "fix" more of a workaround. Windows Speed never really gets matched. 
I just want to mention it so Developers don't assume it's enough just to apply this workaround.
Some Things i observed when comparing Downloadspeeds:
There also seems to be a difference depending on what games are being downloaded. I suspect it to be related to games that have lots of smaller files. The Download Speed of such games is slower and this is somewhat expected and normal behaviour  though the difference in download speed between Windows and Linux (workaround already applied) seems to be even bigger and sometimes just half the speed compared to windows. My guess also is that this is best observed on fast connections. Therefore this workaround might appear to "fix" the issue when used on a slow/average connection. But on faster connections it becomes more obvious that it doesn't really fix the root of the Problem.
I have 3 different systems at home and all the systems only come "close" to Windows (in best case scenarios)  but never match it.

**Comment by tankersss** [comment]
> @nClientDownloadEnableHTTP2PlatformLinux 0
> @fDownloadRateImprovementToAddAnotherConnection 1
> @cMaxInitialDownloadSources 15

It does nothing for me. I used to get up to ~71Mbps (the same max as on windows) but since I updated to Fedora 40 (kernel 6.8.5, since 6.8.9 breaks games) it's irregular (at least it's when I noticed), sometimes full speed, sometimes max 10Mbps. On my Steam Deck wired I still get up to ~71Mbps. Changing server locations does nothing for me too.

**Comment by afonsolage** [comment]
For me, it worked, my download was about ~70Mpbs and after applying those settings, it went up to ~400Mbps

**Comment by Orko79** [comment]
Fix works for me. Download before: ~100 Mbit, after: 1000 Mbit!

**Comment by mferraci** [comment]
Same issue there, and I confirm workaround allowed to jump from 10MB/s to around 95MB/s.

Parameters set: 
`@nClientDownloadEnableHTTP2PlatformLinux 0
@cMaxInitialDownloadSources 15`

**Comment by InferiusFurious** [comment]
This remains an alarming issue, these "fixes" do improve my initial top speed of about ~20Mbps and bumps it to about ~450Mbps, but only solves part of the problem. I can still only get **half of my actual line speed**, which is half of what I can get in Windows.

So to reiterate, in Linux I get around ~450Mbps. Very rarely it will do random bumps up to ~600Mbps, sometimes staying there for a short while before tanking again. I have never seen it go above this though. In Windows I get ~900Mbps+ consistently.

This clearly means both my hardware and my internet is more than capable of downloading at close to 1000Mbps, which means this has **nothing to do with bottlenecked hardware**. This also has **nothing to do with slow Steam download servers**. I put particular emphasis on this because while it can be a factor for some users, this particular issue transcends these cases.

There are too many external forums pointing directly to this issue that seems to blame the user's internet or hardware, and that needs to stop.

I don't have any other issues with any other game clients on Linux (Lutris + EA App goes the full 930Mbps on Linux and same in Windows).

![steam-linux-slow](https://github.com/ValveSoftware/steam-for-linux/assets/113501145/7138be72-a3c2-4093-8338-94f55fe18fe9)
![speedtest-linux-fast](https://github.com/ValveSoftware/steam-for-linux/assets/113501145/1a77729b-f64a-435a-b9c7-7974d5088ab1)
![fast-linux-fast](https://github.com/ValveSoftware/steam-for-linux/assets/113501145/89ced98f-3311-4bbb-bb92-44528f0b889f)

**Comment by chaosmaou** [comment]
I had the same issue here. I got linked to this issue from the ArchWiki steam troubleshooting page. I run the latest EndeavourOS.

I shut down Steam and created the file ```~/.steam/steam/steam_dev.cfg``` with the following contents:

```
@nClientDownloadEnableHTTP2PlatformLinux 0
@cMaxInitialDownloadSources 15
```
Started Steam and my download that was maxing out around 190mbit was now pulling around 931mbit on my gig connection!

**Comment by InferiusFurious** [comment]
> I had the same issue here. I got linked to this issue from the ArchWiki steam troubleshooting page. I run the latest EndeavourOS.
> 
> I shut down Steam and created the file `~/.steam/steam/steam_dev.cfg` with the following contents:
> 
> ```
> @nClientDownloadEnableHTTP2PlatformLinux 0
> @cMaxInitialDownloadSources 15
> ```
> 
> Started Steam and my download that was maxing out around 190mbit was now pulling around 931mbit on my gig connection!

Do you get a consistent 900Mbps+? I am unable to reach that speed on Linux despite having these fixes applied, even though I have a gigabit connection and downloading from the same server I am downloading from on Windows where I easily reach 900Mbps.

**Comment by ItsJustTee** [comment]
> I had the same issue here. I got linked to this issue from the ArchWiki steam troubleshooting page. I run the latest EndeavourOS.
> 
> I shut down Steam and created the file `~/.steam/steam/steam_dev.cfg` with the following contents:
> 
> ```
> @nClientDownloadEnableHTTP2PlatformLinux 0
> @cMaxInitialDownloadSources 15
> ```
> 
> Started Steam and my download that was maxing out around 190mbit was now pulling around 931mbit on my gig connection!

This config worked for me. Using Kubuntu 24.04 and the config fixed my download speed. What was initially a 5.8MB/s download, now got to the maximum ISP allowed Download speed. Thank you so much.

![Screenshot_20240601_140839](https://github.com/ValveSoftware/steam-for-linux/assets/37266538/5f29d3b0-586a-4520-87a1-a5a8eff584ac)

**Comment by chaosmaou** [comment]
> https://github.com/ValveSoftware/steam-for-linux/issues/10248#issuecomment-2143291950

The download speed seems to vary quite a lot. The slowest I have seen it go is usually around the 500-600mbit mark.

I tried running both the steam runtime and steam-native-runtime packages to see if the runtime version was at fault, but using native arch libraries seems to still have the slow downloads.

**Comment by Bill5122** [comment]
Hey guys, ive finally been able to get identical speed to windows doing the following:

i have the following ~/.local/share/Steam/steam_dev.cfg:

@nClientDownloadEnableHTTP2PlatformLinux 0
@fDownloadRateImprovementToAddAnotherConnection 1
@cMaxInitialDownloadSources 15

and ALSO had to blocklist my 'cache1' url, which for me is:

 cache1-pit-pitx.steamcontent.com 

yours is likely to be different, but you can find it by inspecting ~/.local/share/Steam/logs/content_log.txt
go to the bottom of the file, you should see a line similar to the following:

Created download interface of type 'SteamCache' (7) to host cache1-pit-pitx.steamcontent.com (cache1-pit-pitx.steamcontent.com)

You can block this either in your dns settings (this is left as an exercise to the reader) OR by adding the following line to /etc/hosts (substitute my url for yours):

127.0.0.1 cache1-pit-pitx.steamcontent.com

only after applying both of these changes i finally get speed identical to windows

Credits to @roope242 for figuring out the bit about cache1 being weirdly slow

**Comment by mcheman** [comment]
Unfortunately, blocking has not made much of a difference with my setup. I assume it's meant to force a slow server to drop out so a fast one will replace it, however all my download servers contribute roughly equally. 
![equal-bandwidth](https://github.com/ValveSoftware/steam-for-linux/assets/12701028/89721ce2-e658-4f76-941c-d110f23f7f8c)
I've tried blocking more of the slowest servers but the replacements aren't noticeably different. I would need to see ~20MB/s on average from each server to saturate my connection but I only ever see ~7MB/s max.

**Comment by chaosmaou** [comment]
I started diving into and carefully looking through all the steam logs and noticed that when downloading I get thousands of these lines per minute dumped out to `~/.steam/steam/logs/content_log.txt`

```
[2024-06-07 13:57:02] 22633 flagged invalid platform (oslist: linux)
[2024-06-07 13:57:02] 22634 flagged invalid platform (oslist: linux)
[2024-06-07 13:57:02] 22636 flagged invalid platform (oslist: linux)
[2024-06-07 13:57:02] 33245 flagged invalid platform (oslist: windows)
[2024-06-07 13:57:02] 34440 flagged invalid platform (oslist: windows)
[2024-06-07 13:57:02] 34450 flagged invalid platform (oslist: windows)
[2024-06-07 13:57:02] 34460 flagged invalid platform (oslist: windows)
[2024-06-07 13:57:02] 34470 flagged invalid platform (oslist: windows)
[2024-06-07 13:57:02] 34490 flagged invalid platform (oslist: linux)
[2024-06-07 13:57:02] 34491 flagged invalid platform (oslist: linux)
[2024-06-07 13:57:02] 34492 flagged invalid platform (oslist: linux)
[2024-06-07 13:57:02] 34493 flagged invalid platform (oslist: linux)
[2024-06-07 13:57:02] 34494 flagged invalid platform (oslist: linux)
[2024-06-07 13:57:02] 34495 flagged invalid platform (oslist: linux)
[2024-06-07 13:57:02] 34496 flagged invalid platform (oslist: linux)
[2024-06-07 13:57:02] 34497 flagged invalid platform (oslist: linux)
[2024-06-07 13:57:02] 34498 flagged invalid platform (oslist: linux)
[2024-06-07 13:57:02] 35417 flagged invalid platform (oslist: linux)
[2024-06-07 13:57:02] 35419 flagged invalid platform (oslist: linux)
```

It constantly prints this while downloading. I shutdown steam, deleted the file before starting steam back up and more than 4k lines of this is dumped into the file right as steam starts downloading. They don't seem to reappear even if I pause and start new downloads after that.

These messages appear both with and without the steam_dev.cfg file created.

**Comment by scarignan** [comment]
I'm using PoPOS and Steam Flatpack version.

This fixed my issue, thanks for this post!
For people that may be new to linux;

With Steam flatpack version, since its running inside a container the path to edit is here: 

~/.var/app/com.valvesoftware.Steam/.steam/steam

The file "steam_dev.cfg" won't exist at all. I had to created it and added the 3 lines specified, then restarted steam:

@nClientDownloadEnableHTTP2PlatformLinux 0
@fDownloadRateImprovementToAddAnotherConnection 1
@cMaxInitialDownloadSources 15

I went from 100mb/sec to 900mb/sec.

Thanks for this post. Works flawlessly.

**Comment by hasamsote** [comment]
The change in the Reddit post fixes download speeds on my Fedora system. Before I was limited to just about 80 mbps, after I reach my ISP limit which fluctuates around 500 mbps or so.

Pretty sure this problem affects the Steam Deck as well, though I haven't tried applying the fix since it's a bit troublesome. But I did see download speeds drop simultaneously on all my Linux Steam devices to about 10 mbps and this might have been in November or so.

**Comment by ItsJustTee** [comment]
Same behaviour on Debian Gnome without the config file.

**Comment by ItsJustTee** [comment]
Something with the latest Steam Client update messed up the download speeds. With and without the config file, I am getting slower download speeds.

![image](https://github.com/ValveSoftware/steam-for-linux/assets/37266538/7e7db4a3-c692-4b7c-aa1d-50b7b83cb286)

**Comment by epicbattle1** [comment]
> Hey guys, ive finally been able to get identical speed to windows doing the following:
> 
> i have the following ~/.local/share/Steam/steam_dev.cfg:
> 
> @nClientDownloadEnableHTTP2PlatformLinux 0 @fDownloadRateImprovementToAddAnotherConnection 1 @cMaxInitialDownloadSources 15
> 
> and ALSO had to blocklist my 'cache1' url, which for me is:
> 
> cache1-pit-pitx.steamcontent.com
> 
> yours is likely to be different, but you can find it by inspecting ~/.local/share/Steam/logs/content_log.txt go to the bottom of the file, you should see a line similar to the following (you may need to start a download before it will appear):
> 
> Created download interface of type 'SteamCache' (7) to host cache1-pit-pitx.steamcontent.com (cache1-pit-pitx.steamcontent.com)
> 
> You can block this either in your dns settings (this is left as an exercise to the reader) OR by adding the following line to /etc/hosts (substitute my url for yours):
> 
> 127.0.0.1 cache1-pit-pitx.steamcontent.com
> 
> only after applying both of these changes i finally get speed identical to windows
> 
> Credits to @roope242 for figuring out the bit about cache1 being weirdly slow

if anyone's interested, you can add this to dnsmasq's configuration file (/etc/dnsmasq.conf or /etc/NetworkManager/dnsmasq.d/callitwhateveryouwant.conf depending on your setup), it should be universal: `address=/cache[0-9]+-[a-z]+[0-9]+\.steamcontent\.com/127.0.0.1`

**Comment by Skote2** [comment]
![image](https://github.com/user-attachments/assets/1cafcb15-af86-4f9f-8b7a-99a6e3cee69b)

It looks like #11089 might have fixed this. I still have the work around so I'm not entirely certain but the best I could get before that was fixed was ~400mbps and now I'm maxing out my NIC.

**Comment by polluxau** [comment]
can confirm also after removing the cfg file that the speeds are being maxed out on the latest beta of steam

![image](https://github.com/user-attachments/assets/aefdba0d-c112-485e-8fbd-a27227327a5b)

**Comment by ItsJustTee** [comment]
Interesting, so its safe to remove the Config file?

**Comment by polluxau** [comment]
> Interesting, so its safe to remove the Config file?

Well its just gonna recreate the same one so no point

**Comment by ItsJustTee** [comment]
> > Interesting, so its safe to remove the Config file?
> 
> Well its just gonna recreate the same one so no point

Gotcha

**Comment by cchulo** [comment]
> > Interesting, so its safe to remove the Config file?
> 
> Well its just gonna recreate the same one so no point

can someone else confirm? i have not been able to replicate this, i am skeptical that this is true

**Comment by mtorromeo** [comment]
> > > Interesting, so its safe to remove the Config file?
> > 
> > 
> > Well its just gonna recreate the same one so no point
> 
> can someone else confirm? i have not been able to replicate this, i am skeptical that this is true

It's not

**Comment by polluxau** [comment]
> > > > Interesting, so its safe to remove the Config file?
> > > 
> > > 
> > > Well its just gonna recreate the same one so no point
> > 
> > can someone else confirm? i have not been able to replicate this, i am skeptical that this is true
> 
> It's not

It could be because im using cachy os and they have modified the steam client, unsure gonna retest now and check if there are any variables being used in the .desktop or somethin

**Comment by polluxau** [comment]
https://github.com/user-attachments/assets/191fede2-82ac-48ff-adb3-a8a32033eb13

as you can see i delete the file then it gets recreated when i launch steam

**Comment by spiffeeroo** [comment]
@polluxau CachyOS has a patch that adds that steam_dev.cfg file.
https://github.com/CachyOS/CachyOS-PKGBUILDS/blob/ce8434a4edb2ed9d1828c74577e73b56aa2fed16/handheld/steam/deckify.patch#L11

Arch Linux does not have that Steam patch.

**Comment by polluxau** [comment]
> @polluxau CachyOS has a patch that adds that steam_dev.cfg file.
> https://github.com/CachyOS/CachyOS-PKGBUILDS/blob/ce8434a4edb2ed9d1828c74577e73b56aa2fed16/handheld/steam/deckify.patch#L11
> 
> Arch Linux does not have that Steam patch.

Thought so, thanks for the information

**Comment by aidanharris** [comment]
Is anyone from Valve aware of these issues in an official capacity? The tweaks here helped a bit but I'm still getting nowhere near the theoretical maximum for my gigabit connection. The download, although better with these tweaks is not stable at all either it's currently peaking at 48.2 Mbps and the current download speed is all over the place, it will go from 20-30 Mbps right back down to 5 Mbps or 1 or 2 Mbps, etc.

It doesn't matter whether I:
- Change servers between London/Manchester (Manchester seems more stable)
- Change the MTU to various suggested values  (fiddling with that for hours is a lost cause. It does nothing)
- Change the congestion control algorithm from BBR to Cubic
- Disable IPv6 in /etc/hosts: `0.0.0.0 ipv6check-http.steamserver.net` (Not wanting to use the nuclear sysctl option others recommend trying. I quite like IPv6)

It really feels like this is some sort of client issue or the server throttling Linux somehow. All of my other applications have no issue, it's just Steam. I can run a speedtest any time of the day and I'll get roughly the same reported download speed.

**Comment by epicbattle1** [comment]
> Is anyone from Valve aware of these issues in an official capacity? The tweaks here helped a bit but I'm still getting nowhere near the theoretical maximum for my gigabit connection. The download, although better with these tweaks is not stable at all either it's currently peaking at 48.2 Mbps and the current download speed is all over the place, it will go from 20-30 Mbps right back down to 5 Mbps or 1 or 2 Mbps, etc.
> 
> It doesn't matter whether I:
> 
>     * Change servers between London/Manchester (Manchester seems more stable)
> 
>     * Change the MTU to various suggested values  (fiddling with that for hours is a lost cause. It does nothing)
> 
>     * Change the congestion control algorithm from BBR to Cubic
> 
>     * Disable IPv6 in /etc/hosts: `0.0.0.0 ipv6check-http.steamserver.net` (Not wanting to use the nuclear sysctl option others recommend trying. I quite like IPv6)
> 
> 
> It really feels like this is some sort of client issue or the server throttling Linux somehow. All of my other applications have no issue, it's just Steam. I can run a speedtest any time of the day and I'll get roughly the same reported download speed.

https://github.com/ValveSoftware/steam-for-linux/issues/10248#issuecomment-2227480079
see if that works for you

**Comment by lemonase** [comment]
Hello, none of the suggested fixes have changed my download speeds.

I have tried all variations of these settings in my `~/.steam/steam/steam_dev.cfg` file.

```
@nClientDownloadEnableHTTP2PlatformLinux 0
@fDownloadRateImprovementToAddAnotherConnection 1
@cMaxInitialDownloadSources 15
```

Running Arch Linux
Steam version 1722380543

Download speeds are significantly degraded (22Mbps) from speed on Windows (100Mbps+) on the same computer.

**Comment by sleepymincy** [comment]
So, there might be something more to this. I wanna people try doing what I did and see if it shows any difference.

I've tried everything: messing around with dnsmasq, adding all three lines to config files, reinstalling my Wi-Fi dongle drivers, reinstalling Steam (both AUR package and Flatpak had this issue). 

I am on Arch (btw) kernel version `6.10.5-arch1-1`, I use [these drivers](https://github.com/brektrou/rtl8821CU) for my Wi-Fi dongle, `KDE Plasma 6.1.4`. Hardware:

- 12 × 11th Gen Intel® Core™ i5-11400F @ 2.60GHz
- 16GB DDR4
- NVIDIA GeForce RTX 3060
- Z590 VISION G Motherboard

What helped (bootleg really, not actual solution) me is doing USB tether through my phone. Yeah... And before someone will reply to me with "that must be your driver issue", when I do speedtests (using this tool: [speedtest-cli](https://github.com/sivel/speedtest-cli)), it shows similar results both on Wi-Fi and when I tether from phone:

Phone: 
```
❯ speedtest-cli 
Retrieving speedtest.net configuration...
Testing from $MYISP (XXX.XXX.XXX.XXX)...
Retrieving speedtest.net server list...
Selecting best server based on ping...
Hosted by example.com ($MYCITY) [6.13 km]: 8.034 ms
Testing download speed................................................................................
Download: 63.43 Mbit/s
Testing upload speed.....................................................................................
Upload: 63.37 Mbit/s
```

Wi-Fi: 
```
❯ speedtest-cli 
Retrieving speedtest.net configuration...
Testing from $MYISP (XXX.XXX.XXX.XXX)...
Retrieving speedtest.net server list...
Selecting best server based on ping...
Hosted by example.com ($MYCITY) [6.13 km]: 2.661 ms
Testing download speed................................................................................
Download: 61.09 Mbit/s
Testing upload speed.....................................................................................
Upload: 48.75 Mbit/s
```

It's not an ISP issue either. Phone is connected to the same router as the PC, it's just Steam doesn't like being connected through my dongle I guess.

Anyways, Steam results.

Before connecting to Wi-Fi (using USB tethering):

![Screenshot_20240822_054424](https://github.com/user-attachments/assets/b3761456-91cd-4e44-b338-14eeea733cdf)

After connecting to Wi-Fi (no USB tethering):

![Screenshot_20240822_061050](https://github.com/user-attachments/assets/c1915316-515e-40f3-9b06-0957fa9813a2)

**Comment by InferiusFurious** [comment]
For what it's worth, I ended up switching to Nobara Linux and the issue seems to be mostly solved there. I can saturate my 1000Mbps connection.

**Comment by polluxau** [comment]
> For what it's worth, I ended up switching to Nobara Linux and the issue seems to be mostly solved there. I can saturate my 1000Mbps connection.

Doesnt nobara have it enabled by default also?

Thats great tho :)

**Comment by InferiusFurious** [comment]
> > For what it's worth, I ended up switching to Nobara Linux and the issue seems to be mostly solved there. I can saturate my 1000Mbps connection.
> 
> Doesnt nobara have it enabled by default also?
> 
> Thats great tho :)

Not sure what Nobara Linux does by default, all I know is I get these speeds on my gigabit fibre connection, as expected:

![steam-fast-nobara](https://github.com/user-attachments/assets/9caa0f16-c2b4-4138-a56d-89bf11e17fb7)

**Comment by Rosentti** [comment]
Reminder: Steam doesn't download first, install later. Steam uses your CPU and disk while downloading, as the download process is effectively:
Download -> Decompress -> Write. Depending on your disk and CPU, the downloads may be faster or slower. Your CPU and disk scheduler will also cause differences in download speed compared to something that purely downloads, like Speedtest(s). Also, when testing on Windows, make sure to use the same drive for accurate results. 

Also, I feel like other download issues should probably be put into another issue, since this one focuses primarily on HTTP2.

**Comment by InferiusFurious** [comment]
> Also, when testing on Windows, make sure to use the same drive for accurate results.

That is already what a lot of people in this thread are doing. Plenty here are well aware of the CPU requirements to address the overhead required by the decompression algorithm and the write speed of SSDs.

For those of us that have powerful enough gaming computers with high end CPUs and SSDs, it is quite odd that in some Linux distros specifically the Steam download speed drops to half or less than half of our ISP advertised download speed (other apps don't have this issue), whereas on Windows, Steam quite easily saturates the connection using the exact same hardware and decompression algorithms/write speeds.

I personally fixed my slow download speeds by just using Nobara Linux instead (perhaps they mitigate additional HTTP2 issues outside of the steam_dev.cfg file), but your mileage may vary.

**Comment by ryzendew** [comment]
has anyone tested without the config? i have and the issue is gone and has been gone for months now but since everyone is running the fix or a slow cpu they think they still have the issue. 

i think a retest without the fix is needed now.

**Comment by polluxau** [comment]
> has anyone tested without the config? i have and the issue is gone and has been gone for months now but since everyone is running the fix or a slow cpu they think they still have the issue. 
> 
> i think a retest without the fix is needed now.

Oh damn, im running endeavour os and dont have the config applied, i can hit my max download now yay :)

I have a ryzen 7600 also 

Can anyone else check also if we all get the same so we can finally close this?

**Comment by roope242** [comment]
> has anyone tested without the config? i have and the issue is gone and has been gone for months now but since everyone is running the fix or a slow cpu they think they still have the issue.
> 
> i think a retest without the fix is needed now.

Just tested with and without the config, hasn't been fixed. \
The speeds without the fix applied fluctuate alot, with the fix it's very stable.

Running a Ryzen 5800X3D and writing to a WD SN850X. Steam version `1725654351`.

**Comment by cchulo** [comment]
> Running a Ryzen 5800X3D and writing to a WD SN850X. Steam version `1725654351`.

you have the right idea, most replies on here are just complaints about this issue. We need to provide additional info to see if there's a pattern. 

What is your kernel? 'uname -r'
what network card are you using? what motherboard? 
what OS? what version?
and like you posted, what CPU? what storage device? at what speed is the storage device?

**Comment by tonur** [comment]
I also had this issue. No problem on Windows 10, and the suggested fix by the author worked for me.
Writing this comment to report my info in hopes of contributing to investigations of patterns.

<details>
  <summary>Click here for my Linux machines info dump</summary>
  
Steam details:
```
Steam Beta Branch:  Stable Client
Steam Version:  1726604483
Steam Client Build Date:  Tue, Sep 17 10:03 PM UTC -08:00
Steam Web Build Date:  Tue, Sep 17 1:12 AM UTC -08:00
Steam API Version:  SteamClient021
```

```
OS: Manjaro Linux x86_64 
Kernel: 6.10.11-2-MANJARO 
Uptime: 31 mins 
Packages: 1539 (pacman), 20 (flatpak) 
Shell: bash 5.2.32 
DE: Plasma 6.1.5 
WM: KWin  
CPU: AMD Ryzen 5 7600X (12) @ 5.453GHz 
GPU: AMD ATI Radeon RX 6950 XT 
GPU: AMD ATI 0f:00.0 Raphael 
Memory: 9959MiB / 31304MiB 
Motherboard: ASUS ROG STRIX B650E-F GAMING WIFI
```
Storage devices are - nvme1n1: Samsung Evo PCI-E 3.0, nvme0n1: Kingston PCI-E 4.0 
>uname -a: `0a:00.0 Ethernet controller: Intel Corporation Ethernet Controller I225-V (rev 03)`
>lspci: `6.10.11-2-MANJARO #1 SMP PREEMPT_DYNAMIC Thu Sep 26 03:28:26 UTC 2024 x86_64 GNU/Linux`
>lsblk: 
```
NAME        MAJ:MIN RM   SIZE RO TYPE MOUNTPOINTS
loop0         7:0    0 303,4M  1 loop /var/lib/snapd/snap/code/143
loop1         7:1    0  63,5M  1 loop /var/lib/snapd/snap/core20/2015
loop2         7:2    0  40,9M  1 loop /var/lib/snapd/snap/snapd/20290
nvme1n1     259:0    0 953,9G  0 disk 
├─nvme1n1p1 259:1    0   100M  0 part /boot/efi
├─nvme1n1p2 259:2    0    16M  0 part 
├─nvme1n1p3 259:3    0 470,1G  0 part 
├─nvme1n1p4 259:4    0   530M  0 part 
└─nvme1n1p5 259:5    0 483,1G  0 part /
nvme0n1     259:6    0   1,8T  0 disk 
└─nvme0n1p1 259:7    0   1,8T  0 part /run/media/<USER>/SSD
```

</details>

**Comment by lostgoat** [comment]
> has anyone tested without the config? i have and the issue is gone and has been gone for months now but since everyone is running the fix or a slow cpu they think they still have the issue.
> 
> i think a retest without the fix is needed now.

A fix for the HTTP2 issue was deployed server side around a month and a half ago.

**Comment by agurenko** [comment]
I can confirm from my side as well, I've been running without `steam_dev.cfg` for several weeks and while the speed is not ramping up as fast as with config, it does go above 100 MB/s on my 1 Gbps connection.

**Comment by LingMan** [comment]
Same here. Works without the `steam_dev.cfg` now.

**Comment by polluxau** [comment]
Same here, I got an upgrade in speed and steam can reach it pretty fast without the config, I'll be closing this issue soon, so if anyone has issues still please speak up 🙃

**Comment by Orko79** [comment]
> Fix works for me. Download before: ~10 MB/s, after: 100 MB/s!

Hi! I recently tested without the fix. Can confirm that it is not needed any longer. Anyone should disable it. At least in my case, the download speed is higher without the fix now!

**Comment by andythompson** [comment]
This same issue started for me a few weeks ago and is still happening. I have tried the config in the original post, variations and no steam_dev.cfg.

I max out at about 50mbits on a symmetrical 1gb connection. My Steam Deck has no issues with wired or wifi and usually gets 600mbits+.

<details>

<summary>System details here</summary>

```bash
uname -a
Linux arch 6.11.3-arch1-1 #1 SMP PREEMPT_DYNAMIC Thu, 10 Oct 2024 20:11:06 +0000 x86_64 GNU/Linux
Model name:                           12th Gen Intel(R) Core(TM) i9-12900K

lspci | grep -i '\(vga\|eth\)'
01:00.0 VGA compatible controller: NVIDIA Corporation AD103 [GeForce RTX 4080 SUPER] (rev a1)
06:00.0 Ethernet controller: Intel Corporation Ethernet Controller I225-V (rev 03)

```

```bash
Steam Version: 1726604483
```

```bash
2: enp6s0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP group default qlen 1000

speedtest-cli
Testing download speed................................................................................
Download: 657.45 Mbit/s
Testing upload speed......................................................................................................
Upload: 821.04 Mbit/s
```

</details>

**Comment by chaosmaou** [comment]
I recently upgraded to a whole new system and did a clean install of EndeavourOS. I am now getting around 77-78 MB/s, with the limiter set to 80 MB/s just like in Windows. It's possible that newer packages have fixed this issue, as it's much better now without any tweaking.

**Comment by hpeter1994** [comment]
Removed the config and my system managed to do 100MB/s without it. I also dont need the DNS caching server anymore.
Issue seems to be fixed for now.

**Comment by HeeroML** [comment]
> Same here, I got an upgrade in speed and steam can reach it pretty fast without the config, I'll be closing this issue soon, so if anyone has issues still please speak up 🙃

Had the issue today. Fixed with config

**Comment by breadland** [comment]
On Arch Linux at least, I'm getting 600-900mbps download speeds WITHOUT the config.
I'd say it's about as good as Windows now.

**Comment by polluxau** [comment]
> > Same here, I got an upgrade in speed and steam can reach it pretty fast without the config, I'll be closing this issue soon, so if anyone has issues still please speak up 🙃
> 
> Had the issue today. Fixed with config

What distro and hardware?

**Comment by Justin-Hoffman** [comment]
I still see this on Fedora 39. I have found that in general it's fastest to download games to my steam deck (which doesn't seem to have this issue) then transfer locally over the network to my other device (which, frustratingly, still isn't as fast as the download from internet to the deck).

**Comment by saviit** [comment]
Still not resolved, I'm still getting download speeds of 30-50 Mbit/s on a 100 Mbit/s connection. None of the proposed workarounds in this issue have had any effect on this whatsoever.

Distro: Arch Linux, kernel 6.11.6-arch1-1 (fresh install)
System:
AMD 7950X
Asus TUF X670E Plus
64 GB RAM
WD SN850X 2TB

Steam Beta Branch:  Steam Beta Update
Steam Version:  1731112063
Steam Client Build Date:  Sat, Nov 9 01:37 UTC -08:00
Steam Web Build Date:  Sat, Nov 9 01:55 UTC -08:00
Steam API Version:  SteamClient021

tldr workarounds tried:
- changing steam client branch had no effect
- changing download region had no effect
- changing router's DNS server between ISP default and 1.1.1.1 had no effect
- steam_dev.cfg tweaks did not work (but see below)
- dnsmasq config had no effect
- not locked to only one server, so using `/etc/hosts` to block unwanted servers (see https://github.com/ValveSoftware/steam-for-linux/issues/9715#issuecomment-2425357737) is pointless

I still have a completely separate Windows (11, 22H2) installation on the same system (not dual boot; I boot from UEFI if I need to use Windows). On Windows I don't have this issue, downloads are always on 100% of bandwidth.

I'm not locked to only one download server (without any tweaks):
![dl_sources_default](https://github.com/user-attachments/assets/9622b6ea-99a5-4034-ae2b-c5ed52bfe002)

The proposed HTTP2/download server tweaks with `steam_dev.cfg` do allow more connections, but it has absolutely no effect on my actual download speed:
![dl_sources](https://github.com/user-attachments/assets/c7b2c4b8-d8b3-4d6f-93d0-e7fc51245103)

**Comment by kisak-valve** [comment]
Hello @saviit, your symptoms don't match the issue that was tracked in this issue report. Please open a new issue report.

**Comment by andythompson** [comment]
Update: switching to the beta version fixed the issue for me. No other changes applied. 

https://github.com/ValveSoftware/steam-for-linux/issues/10248#issuecomment-2408743529

**Comment by saviit** [comment]
> Hello @saviit, your symptoms don't match the issue that was tracked in this issue report. Please open a new issue report.

New issue posted #11442, in case anyone else still has the same issue.

**Comment by MrNessuno** [comment]
Today I tested the:
@nClientDownloadEnableHTTP2PlatformLinux 0

before I had 25MB/s and after i got over 70MB/s 
is the issue still here?
OS: Manjaro unstable branch (but is months that I noticed that)
with Proton VPN

**Comment by lostgoat** [comment]
@MrNessuno `nClientDownloadEnableHTTP2PlatformLinux 0` should be the default for linux clients.

**Comment by MrNessuno** [comment]
> [@MrNessuno](https://github.com/MrNessuno) `nClientDownloadEnableHTTP2PlatformLinux 0` should be the default for linux clients.

Well, probably isn't because it was night and day difference.

**Comment by grzyvva** [comment]
Using nClientDownloadEnableHTTP2PlatformLinux 0 on flatpak steam client definitely helped.
After changing this download speeds are more in line with how it downloads on my windows laptop.

**Comment by aidanharris** [comment]
I'm still seeing this issue with poor download speeds from various servers. I live in the UK so switch servers a lot when this happens, from UK (London) to New York or France (Paris), eventually Netherlands worked. Please implement a way to use servers from multiple regions. There is not a huge difference in latency in terms of downloading from London, New York, Paris or Netherlands so why is there no option to download from all of Europe or Europe and East Coast US? It doesn't make sense to force the use of a single regions servers.

![Image](https://github.com/user-attachments/assets/128a2189-600a-402d-b523-0987bc7e6a95)

**Comment by lostgoat** [comment]
The issue described in this bug has been addressed. If you are running into a new issue with similar symptoms please open a separate issue on github and include the logs referenced by the issue template.

**Comment by Reonu** [comment]
I don't think this issue has been solved. I was experiencing very slow downloads (~15 MB/s with random, erratic variation). Once I disabled HTTP2 via the .cfg file by using `@nClientDownloadEnableHTTP2PlatformLinux 0` the speeds immediately started to saturate my 600 Mbit connection, downloading at upwards of 70 MB/s. Enabling HTTP2 again makes it slow again. I'm not changing anything other than the HTTP2 setting. 

Sometimes downloads start strong with HTTP2 enabled, but it will always die after a few seconds and start being slow. It will randomly shoot back up and saturate my connection for a while sometimes but it's always erratic. Only with HTTP2 disabled can I maintain my connection saturated indefinitely.

The behavior is exactly the same as what I was experiencing before this issue was "resolved".

I have attached my logs while the client is downloading with HTTP2 enabled.

[logs.tar.gz](https://github.com/user-attachments/files/22688085/logs.tar.gz)

**Comment by Maelstromeous** [comment]
I can confirm this is not fully resolved. I am getting 500mbits on a 900mbit connection, only on Linux (CachyOS). MTU is 1500, disk is an NVME SSD rated at about 1.2GB/s so it's deffo not disk.

**Comment by pollux78** [comment]
> I can confirm this is not fully resolved. I am getting 500mbits on a 900mbit connection, only on Linux (CachyOS). MTU is 1500, disk is an NVME SSD rated at about 1.2GB/s so it's deffo not disk.

As I said in the main post, create a new issue, I can peak my speeds at 400 to 500 without problems with no tweaks needed, this could be a seperate problem now as many users don't have issues with it.

**Comment by Krutonium** [comment]
Honestly even with `@nClientDownloadEnableHTTP2PlatformLinux 0` I typically only a fraction of my total download speed - but it's a much bigger fraction.

(~500Mbps instead of ~15Mbps, on a 1Gbps connection)

**Comment by sof006** [comment]
Made a github account just to comment that I am also having this issue, I have a 2.5Gig line (up and down) and everywhere else I can hit my full speed however in steam at most all I get is 100MB's (if I am lucky). If I use EA App or similar its always full speed.

On Windows this is never an issue, regardless of what I use. Steam on Linux is very much an issue when it comes to downloads.

**Comment by breningham** [comment]
Hi, just thought i'd throw my hat into the ring - this is still an issue on Bazzite, without using `@nClientDownloadEnableHTTP2PlatformLinux 0` max download speeds was ~560Mbps, with `@nClientDownloadEnableHTTP2PlatformLinux 0` i get peaks of 1Gbps (averages around 700Mbps)

**Comment by Reverrb** [comment]
> This only "improves" download speed on my end. but it still isn't able to reach the full potential i get when using Windows. There is something wrong. And all the fixes out there only seem to "improve". Though if you have a somewhat fast connection you still might notice slower speeds than possible. For example i never get my 1Gbps i only get around 700Mbps.. if i disable Https2.. if i don't disable it i will be at around 300Mbps.. while in Windows I'm at pretty much 900+Mbps. I assume lots of people who claim this fixes the issue don't have such a fast connection therefore it already maxes speed after that fixes.

I totally agree with this and some of the other people here. This is more of a treatment rather than a cure.

I assume that valve may not be paying much attention to this because obviously their main player base is windows, but it's likely also because people that are using linux don't have connections faster than 500mbps, especially with the WIFI-only steam deck, meaning they won't see that they're being throttled. However, when you see the issues from people with over 1Gb plans, we are the only ones who can see it.

**Comment by vulptex** [comment]
might be of interest:

https://github.com/ValveSoftware/steam-for-linux/issues/12082#issuecomment-4745731646
